src/types/JSONRPC.d.ts

```diff
@@ -0,0 +1,36 @@
+declare namespace JSONRPC {
+  type JSONRPC = '2.0';
+  type JSONRPCID = string | number | null;
+  type JSONRPCParams<T = any> = T;
+
+  interface JSONRPCError {
+    code: number;
+    message: string;
+    data?: any;
+  }
+
+  interface JSONRPCRequest<TParams = any> {
+    jsonrpc: JSONRPC;
+    method: string;
+    params?: JSONRPCParams<TParams>;
+    id?: JSONRPCID;
+  }
+
+  interface JSONRPCSuccessResponse<TResult> {
+    jsonrpc: JSONRPC;
+    id: JSONRPCID;
+    result: TResult;
+    error?: undefined;
+  }
+
+  interface JSONRPCErrorResponse {
+    jsonrpc: JSONRPC;
+    id: JSONRPCID;
+    result?: undefined;
+    error: JSONRPCError;
+  }
+
+  type JSONRPCResponse<TResponse = any> =
+    | JSONRPCSuccessResponse<TResponse>
+    | JSONRPCErrorResponse;
+}
```

Comment:
Can u try ur best to replace `any`

Reply:
There already has namespace(prefix) `JSONRPC`
No need to put prefix `JSONRPC` on name of every type

Reply:
This kind of file naming reference
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html